go.mod

```diff
@@ -115,34 +120,34 @@ require (
 	go.opencensus.io v0.24.0 // indirect
 	go.uber.org/multierr v1.11.0 // indirect
 	golang.org/x/crypto v0.14.0 // indirect
-	golang.org/x/exp v0.0.0-20220722155223-a9213eeb770e // indirect
 	golang.org/x/net v0.17.0 // indirect
-	golang.org/x/oauth2 v0.8.0 // indirect
+	golang.org/x/oauth2 v0.10.0 // indirect
 	golang.org/x/sys v0.13.0 // indirect
 	golang.org/x/term v0.13.0 // indirect
 	golang.org/x/text v0.13.0 // indirect
+	golang.org/x/xerrors v0.0.0-20220907171357-04be3eba64a2 // indirect
 	gomodules.xyz/jsonpatch/v2 v2.4.0 // indirect
 	google.golang.org/appengine v1.6.7 // indirect
-	google.golang.org/genproto/googleapis/rpc v0.0.0-20230525234030-28d5490b6b19 // indirect
-	google.golang.org/protobuf v1.30.0 // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20230530153820-e85fd2cbaebc // indirect
+	google.golang.org/protobuf v1.31.0 // indirect
 	gopkg.in/go-playground/validator.v9 v9.31.0 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/ini.v1 v1.62.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	k8s.io/component-base v0.28.3 // indirect
-	k8s.io/klog/v2 v2.100.1 // indirect
+	k8s.io/klog/v2 v2.110.1 // indirect
 	k8s.io/kube-aggregator v0.26.1 // indirect
-	k8s.io/kube-openapi v0.0.0-20230717233707-2695361300d9 // indirect
+	k8s.io/kube-openapi v0.0.0-20231010175941-2dd684a91f00 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
 	sigs.k8s.io/kube-storage-version-migrator v0.0.4 // indirect
-	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
 // These replace stanzas are necessary to import maistra/istio-operator and
 // github.com/operator-framework/operator-sdk.
 replace (
 	bitbucket.org/ww/goautoneg => github.com/munnerz/goautoneg v0.0.0-20120707110453-a547fc61f48d
-	github.com/openshift/api => github.com/openshift/api v0.0.0-20231113114413-39964e6af314
-	k8s.io/client-go => k8s.io/client-go v0.28.0
+	github.com/openshift/api => github.com/openshift/api v0.0.0-20240126160819-e54a015ffc8c
+	k8s.io/client-go => k8s.io/client-go v0.29.0
```

Comment:
I don't think we need to or should rebase the K8S api changes on a backport. `release-4.15` should be 0.28 packages.

Reply:
Moving k8s.io/client-go's version back to v0.28.0 were resulting in these `verify` failures. https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-ingress-operator/1022/pull-ci-openshift-cluster-ingress-operator-release-4.15-verify/1753142958116311040

Reply:
Did you roll back `k8s.io/api v0.29.0` and `k8s.io/apimachinery v0.29.0` as well? That passed verify for me locally. Sorry I meant all K8S packages.

I recommend to start with a clean `go.mod` from the latest commit in release-4.15 without your changes, then update `github.com/openshift/api => github.com/openshift/api v0.0.0-20240126160819-e54a015ffc8c` only and do `go mod tidy` and `go mod vendor`.

You'll get a much cleaner `go.mod`, I only see one other dependency bump (`golang.org/x/exp`). We should try to only bump with is specifically required for limiting the impact of this backport.

Reply:
Updated with minimal changes to go.mod.